gems/aws-sdk-core/lib/aws-sdk-core/shared_config.rb

```diff
@@ -389,7 +385,7 @@ def sso_credentials_from_profile(cfg, profile)
           sso_role_name: prof_config['sso_role_name'],
           sso_session: prof_config['sso_session'],
           sso_region: sso_region,
-          sso_start_url: prof_config['sso_start_url']
+          sso_start_url: sso_start_url
```

Comment:
This change should be safe since the non-legacy flow does NOT use the sso_start_url, see: https://github.com/aws/aws-sdk-ruby/blob/version-3/gems/aws-sdk-core/lib/aws-sdk-core/sso_credentials.rb#L82